InvenTree/plugin/registry.py

```diff
@@ -451,7 +456,12 @@ def safe_reference(plugin, key: str, active: bool = True):
             plg_key = slugify(plg.SLUG if getattr(plg, 'SLUG', None) else plg_name)  # keys are slugs!
 
             try:
-                plg_db, _created = PluginConfig.objects.get_or_create(key=plg_key, name=plg_name)
+                if plg_key in plugin_configs:
+                    # Configuration already exists
+                    plg_db = plugin_configs[plg_key]
```

Comment:
Question: Have you tested that changes in the DB are loaded this way on a "soft restart" or reload?

Reply:
Yes, seems to work just fine on "soft" reload